### ISSUE TITLE
Improve ThreadDump by adding waiting method

### ISSUE THREAD

**Issue by minborg** [OpenHFT/Chronicle-Core#96]
Some tests using ThreadDump could be substantially sped up if there was a method that waited for all required threads to close. This would prevent tests from having to wait an arbitrary time before checking thread consistency potentially waiting unnecessarily long.

**Comment by minborg** [comment]
For example
```
final class FooTest {

    @Test
    void bar() {
          ThreadDump threadDump = new ThreadDump();
          Thread t = new Thread(....);
          t.start();
          // Will at most wait 10 seconds
          threadDump.assertNoNewThreads(10, TimeUnit.SECONDS);
     }

}
```

Instead of:

```
final class FooTest {

    @Test
    void bar() {
          ThreadDump threadDump = new ThreadDump();
          Thread t = new Thread(....);
          t.start();
          // Will always wait 10 seconds
          Jvm.pause(TimeUnit.SECONDS.toMillis(10));
          threadDump.assertNoNewThreads();
     }

}

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.20.102](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.20.102), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
